packages/wallet/src/redux/protocols/advance-channel/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,39 @@
+import * as states from '../states';
+import { initialize, reducer } from '../reducer';
+import * as scenarios from './scenarios';
+import { CommitmentType } from '../../../../domain';
+import { expectTheseCommitmentsSent } from '../../../__tests__/helpers';
+
+const itTransitionsTo = (
+  result: states.AdvanceChannelState,
+  type: states.AdvanceChannelStateType,
+) => {
+  it(`transitions to ${type}`, () => {
+    expect(result.type).toEqual(type);
+  });
+};
+
+describe('sending preFundSetup as A', () => {
+  const scenario = scenarios.newChannelAsA;
+
+  describe.only('when initializing', () => {
```

Comment:
The tests are not fully written, but I think the scenarios are mostly complete.

Only this test is enabled.